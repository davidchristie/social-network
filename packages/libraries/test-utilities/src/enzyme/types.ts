import { ReactWrapper, ShallowWrapper } from "enzyme";

export type Wrapper = ReactWrapper | ShallowWrapper;
