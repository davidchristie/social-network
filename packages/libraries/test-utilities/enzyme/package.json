{
  "main": "../dist/enzyme/index.js",
  "name": "test-utilities/enzyme",
  "private": true,
  "version": "1.0.0"
}
