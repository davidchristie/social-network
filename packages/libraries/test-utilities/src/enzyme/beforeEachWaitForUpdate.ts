import { Wrapper } from "./types";

export default function beforeEachWaitForUpdate (getWrapper: () => Wrapper) {
  beforeEach(async () => {
    await new Promise(resolve => window.setTimeout(resolve, 0));
    getWrapper().update();
  });
}
