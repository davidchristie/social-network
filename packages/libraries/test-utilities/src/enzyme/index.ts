export { default as beforeEachWaitForUpdate } from "./beforeEachWaitForUpdate";
export { default as itContainsComponent } from "./itContainsComponent";
