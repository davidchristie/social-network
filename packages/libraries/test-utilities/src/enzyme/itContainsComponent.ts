import React from "react";
import { Wrapper } from "./types";

export default function itContainsComponent (
  name: string,
  getWrapper: () => Wrapper,
  Component: React.ComponentType
) {
  it(name, () => {
    expect(getWrapper().find(Component).exists()).toBe(true);
  });
}
