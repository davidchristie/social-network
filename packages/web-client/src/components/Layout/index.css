.Layout {
  background: var(--theme-secondary, white);
  color: var(--theme-secondary-contrast, black);
  min-height: 100vh;
}

.Layout .page-content {
  padding-top: var(--theme-header-height, 30px);
}
