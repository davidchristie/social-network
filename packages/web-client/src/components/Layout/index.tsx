import React from "react";

import Container from "../Container";
import Header from "../Header";
import "./index.css";

export default class Layout extends React.Component {
  public render () {
    return (
      <div className="Layout">
        <Header />
        <div className="page-content">
          <Container>
            {this.props.children}
          </Container>
        </div>
      </div>
    );
  }
}
