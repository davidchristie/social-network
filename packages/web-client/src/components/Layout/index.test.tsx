import { shallow, ShallowWrapper } from 'enzyme'
import React from 'react'

import Layout from './index'

describe('Layout component', () => {
  let wrapper: ShallowWrapper

  beforeEach(() => {
    wrapper = shallow(<Layout />)
  })

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
