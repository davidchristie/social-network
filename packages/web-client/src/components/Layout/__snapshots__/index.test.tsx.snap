// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component matches snapshot 1`] = `
<div
  className="Layout"
>
  <Header />
  <div
    className="page-content"
  >
    <Container />
  </div>
</div>
`;
